### PR TITLE
Closes #2305: inline JSONArray.mapNotNull

### LINKED ISSUE #2305: Inline `JSONArray.mapNotNull` when android gradle plugin 3.4.0 is released
From https://github.com/mozilla-mobile/android-components/pull/2279#issuecomment-471133448:

> This issue is caused because the google android gradle plugin instruments methods before they are inlined, which breaks jacoco when testing across module boundaries: https://issuetracker.google.com/issues/109771903 (and jacoco [jacoco/jacoco#688](https://github.com/jacoco/jacoco/issues/688) which was closed in favor of the google issue). This is fixed in 3.4.0 (current is 3.3.0, released in January).
> 
> The function is inlined for performance (like Collection.map) so the solution is to temporarily uninline it and wait for the version bump.

We should inline `JSONArray.mapNotNull` when gradle gradle plugin 3.4.0 is released.

### PR DESCRIPTION
**Pull request by mcomella**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/org/json/JSONArray.kt [hunk 1]
```diff
@@ -39,7 +39,6 @@ fun <T> JSONArray?.toList(): List<T> {
     return listOf()
 }
 
-// #2305: inline this function when Android gradle plugin v3.4.0 is released
```
Comment (pocmo) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/org/json/JSONArray.kt]: 3.4.0 is released but AC is on 3.3.x at the moment.

**Comment by pocmo** [comment]
We are on 3.4.1 now, I'll retrigger the task.

**Comment by mcomella** [comment]
@pocmo Can this be merged?

**Comment by mcomella** [comment]
I rebased my changes on master, in particular, the changelog changes which were very out-dated.

**Pull request merged**